mp/src/game/server/momentum/mom_system_saveloc.cpp

```diff
@@ -260,11 +260,49 @@ void CSaveLocSystem::LevelInitPreEntity()
     }
 }
 
+bool CSaveLocSystem::LoadStartMarks()
+{
+    if (m_pSavedLocsKV)
+    {
+        DevLog("Loaded startmarks from %s!\n", SAVELOC_FILE_NAME);
+
+        if (!m_pSavedLocsKV->IsEmpty())
```

Comment:
All of these "cascading ifs" should be early returns where possible. The goal is to make all of your validations be up top, getting the function out early upon failure of some validator and keeping the code left-aligned (and muuuch easier to follow) as much as possible.

```
if (someValidationFails)
   return;

 . . .

if (someOtherValidationFailsThatReliesOnTheFirstOne)
    return;

. . . 

( actual processing code knowing that there's no nulls or whatever )
```

versus

```
if (check1)
{
     if (check2)
     {
            if (check3)
            {
                   . . .

                   etc
```

Reply:
Nested ifs have now been changed to early returns if possible.